vectordraw/vectordraw.py

```diff
@@ -0,0 +1,312 @@
+        scope=Scope.content
+    )
+
+    expected_result = String(
+        display_name="Expected result",
+        help=(
+            "Defines vector properties for grading. "
+            "Vectors omitted from this setting are ignored when grading."
+        ),
+        default="{}",
+        multiline_editor=True,
+        resettable_editor=False,
+        scope=Scope.content
+    )
+
+    custom_checks = String(
+        display_name="Custom checks",
+        help=(
+            'List of custom checks to use for grading. '
+            'This is needed when grading is more complex and cannot be defined in terms of "Expected results" only.'
+        ),
+        default="[]",
+        multiline_editor=True,
+        resettable_editor=False,
+        scope=Scope.content
+    )
+
+    weight = Float(
+        display_name="Weight",
+        default=1,
+        scope=Scope.settings,
+        enforce_type=True
+    )
+
+    # User state
+    answer = Dict(scope=Scope.user_state)
+    result = Dict(scope=Scope.user_state)
+
+    editable_fields = (
+        'display_name',
+        'description',
+        'width',
+        'height',
+        'bounding_box_size',
+        'axis',
+        'show_navigation',
+        'show_vector_properties',
+        'show_slope_for_lines',
+        'add_vector_label',
+        'vector_properties_label',
+        'background_url',
+        'background_width',
+        'background_height',
+        'vectors',
+        'points',
+        'expected_result',
+        'custom_checks'
+    )
+
+    has_score = True
+
+    @property
+    def settings(self):
+        return {
+            'width': self.width,
+            'height': self.height,
+            'bounding_box_size': self.bounding_box_size,
+            'axis': self.axis,
+            'show_navigation': self.show_navigation,
+            'show_vector_properties': self.show_vector_properties,
+            'show_slope_for_lines': self.show_slope_for_lines,
+            'add_vector_label': self.add_vector_label,
+            'vector_properties_label': self.vector_properties_label,
+            'background': self.background,
+            'vectors': self.vectors_json,
+            'points': self.points_json,
+            'expected_result': self.expected_result_json
+        }
+
+    @property
+    def user_state(self):
+        user_state = self.answer
+        if self.result:
+            user_state['result'] = self.result
+        return user_state
+
+    @property
+    def background(self):
+        return {
+            'src': self.background_url,
+            'width': self.background_width,
+            'height': self.background_height,
+        }
+
+    @property
+    def vectors_json(self):
+        return json.loads(self.vectors)
+
+    @property
+    def points_json(self):
+        return json.loads(self.points)
+
+    @property
+    def expected_result_json(self):
+        return json.loads(self.expected_result)
+
+    def resource_string(self, path):
```

Comment:
You probably don't need this function when using the `ResourceLoader` class from xblock-utils.